neuralmonkey/runners/base_runner.py

```diff
@@ -25,16 +25,20 @@ def collect_results(self, results: List[Dict]) -> None:
 
 def collect_encoders(coder):
```

Comment:
Toho je potřeba se zbavit. Každý `ModelPart` bude mít metodu `get_dependencies`, které a tyhle metody to rekuretně posbírají. Pokud se ti to nechce dělat v tomhletom PR, udělej na to issue.

Reply:
Nemohly by prostě ty části rovnou volat feed dict svejch dependencís?

Reply:
je to Dict, takže zavolá-li se náhodou znova, tak se snad nic hroznýho nestane, ne?

Reply:
Radši bych to nechal s těma setama a volal na všem feed dict právě jednou. Když bude na vstupu obrázek nebo signál, tak by se tam mohlo dít nějaké počítání navíc.